### PR TITLE
LPS-146488 LPS-146490 LPS-146489 [CP 2.0] Team invitation integration with koroneiki

### PR DESCRIPTION
**Pull request by joaosvassis**
### How it was developed?
It was added a new GraphQL query to create/update an user associating with an account (project) and also associate a accountRole for this user in an account (project).

I added the `associateUserAccountWithAccountAndAccountRole` GraphQL query that run two mutations (eaedb544f65ecf16404521a9dd146fa604f54f7d).
```
export const associateUserAccountWithAccountAndAccountRole = gql`
	mutation associateUserAccountWithAccountAndAccountRole(
		$emailAddress: String!
		$accountKey: String!
		$accountRoleId: Long!
	) {
		createAccountUserAccountByExternalReferenceCodeByEmailAddress(
			emailAddress: $emailAddress
			externalReferenceCode: $accountKey
		)
		createAccountByExternalReferenceCodeAccountRoleUserAccountByEmailAddress(
			accountRoleId: $accountRoleId
			emailAddress: $emailAddress
			externalReferenceCode: $accountKey
		)
	}
`;
```

_In my test these mutation don't run in same time._

- `createAccountUserAccountByExternalReferenceCodeByEmailAddress`: Create/Update an user by e-mail and associate with an account (project).
  - For new user, it's create with first and last name filled with e-mail. I asked Amos about this behavior, and this is not a problem because the Okta replace these infos with the data come from raysource (after first login).
- `createAccountByExternalReferenceCodeAccountRoleUserAccountByEmailAddress`: Associate a accountRole of a project for the user.

This query was used within a `Promise.all` that run it for each user selected in form (b39db6c9c0f8b794885349dc3e9fab1e2325dd4a).
Maybe in the future we can change it for a batch request. I saw that has for these request, I choose for don't use them because was been necessary a huge changes.

Basically was this, but it was necessary refactor the `InviteTeamMembersForms` component to get both id and name from select role.
Another change for this component was related with the use of roles
- Before was just necessary treat with string or roles' names, but now it's necessary treat the role as entity, in other words, it's necessary use the id and name.
  - For fix this, I add an `onChange` handler for selects inputs that find the role entity by the id selected.
- On component render start, there are some roles business to choose if some roles appears or not. It was necessary also change it, becase before it was only necessary check if some roles name exist and if not, just added. But as now we are treat as entity, it's was necessary remove the entire role.
  - Because this, was necessary some changes in site-initializer backend to add new roles that can be remove or keep it (https://github.com/pt-liferay-solutions/liferay-portal/pull/849).

**Endpoint**
  - To run the endpoint, a new rest was added to fill in the user's accountRole.
imgIgor1111

### What has been affected by the development (optional)?

All flows that use `InviteTeamMembersForms` component was affected
- Onboarding ([LPS-140151](https://issues.liferay.com/browse/LPS-140151))
- Modal in Team Members section ([LPS-140195](https://issues.liferay.com/browse/LPS-140195))

**Endpoint**
   - Had to create logic to get sessionId to use the rest api, affected the onboarding context and the flows that use the InviteTeamMembersForms component.
img_session1

### Test Plan

1. Go to Team Members Form
  - By Onboarding flow:
    - The user logged need be an Account Administrator for the project.
    - Check if doesn't have AccountFlags entries for the project that you want to access the Onboarding flow.
    - Select some project that is associated with Garretz, and came from AccountSync, click on the project and you will be taken to Onboarding Flow, the second step is the form.
  - By Team Members section
    - Select any project that is associated with Garretz, and that came from AccountSync, and go to the Team Member section by selecting from the side menu
    - Click on `+ Invite` button.
2. Fill out with garrettz@mbopartners.com.broken and select any roles and click on `Send Invitations`.
3. Go to the Users list for Account for this project.
  - Control Panel > Account > Find the account by the External Reference Code > Click on User tab
4. Check if the users that was filled in the form appears here, with the roles that was choose for them.

### It's necessary to add data mocks (optional)?
No.

### CODE REVIEW

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js [hunk 1]
```diff
@@ -67,183 +87,170 @@ const InvitesPage = ({
...
 
 			setAvailableAdminsRoles(remainingAdmins);
 		}
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [values, project, rolesData]);
+	}, [values, project, maxRequestors, accountRoles]);
 
 	useEffect(() => {
+		const filledEmails =
+			values?.invites?.filter(({email}) => email)?.length || 0;
```
Comment (claricebrasil) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js:191]: Do we need this {} around the email here?

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js [hunk 2]
```diff
@@ -67,183 +87,170 @@ const InvitesPage = ({
...
 		}
 		else if (touched['invites']?.some((field) => field?.email)) {
 			setInitialError(true);
 			setBaseButtonDisabled(true);
 		}
-	}, [touched, failedEmails, filledEmails, totalEmails]);
+	}, [touched, values, errors]);
+
+	const handleSubmit = async () => {
+		const filledEmails = values?.invites?.filter(({email}) => email) || [];
```
Comment (claricebrasil) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js:209]: Do we need this {} around email?

**Review by igordmf — APPROVED** [review]
Great job!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js [hunk 3]
```diff
@@ -210,6 +210,24 @@ export const addTeamMembersInvitation = gql`
...
+	mutation associateUserAccountWithAccountAndAccountRole(
+		$emailAddress: String!
+		$accountKey: String!
+		$accountRoleId: Long!
+	) {
+		createAccountUserAccountByExternalReferenceCodeByEmailAddress(
+			emailAddress: $emailAddress
+			externalReferenceCode: $accountKey
+		)
+		createAccountByExternalReferenceCodeAccountRoleUserAccountByEmailAddress(
```
Comment (claricebrasil) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js:223]: this variables names has to be this size or we can use something more generic like creatAccountByEmailAddress? It seems a little bit confusing and hard to read.

**Review by claricebrasil — APPROVED** [review]
Great job, Joao! I missed some clarity in your description. Maybe you can explain more what you did in terms of usability, I thought that for someone who doesn´t know what the activity is all about, it`s a little hard to understand what you did. Maybe you can put some visual stuff or something. Let me know if you need any help. About the code, congratulations! It`s great!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js (hunk 1 above)
Comment (jairmedeiros) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js:191]: yeah. The invite is a object with email and role properties. Using the {} it's for use only the email value

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js (hunk 2 above)
Comment (jairmedeiros) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js:209]: yeah. The invite is a object with email and role properties. Using the {} it's for use only the email value

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js (hunk 3 above)
Comment (jairmedeiros) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/services/liferay/graphql/queries.js:223]: Unfortunately this is a request, not a variable haha. We need to use this huge sentence 😢

**Review by joanagrego — APPROVED** [review]
Nice job!

**Comment by jairmedeiros** [comment]
ci:forward

**Comment by jairmedeiros** [comment]
ci:forward

**Review by Lucas2691 — APPROVED** [review]
Gob job!

**Comment by joaosvassis** [comment]
ci:forward

**Review by thaynam — APPROVED** [review]
Good job!

**Comment by joaosvassis** [comment]
ci:forward

**Comment by jairmedeiros** [comment]
Just starting reviewing :)

**Comment by iagochaves** [comment]
ci:forward

**Comment by joaosvassis** [comment]
Just starting reviewing :)

**Pull request closed without merging**